File: src/main/java/dev/tuhkanens/comfortlib/api/MessagesAPI.java
package dev.tuhkanens.comfortlib.api;

import net.kyori.adventure.text.minimessage.tag.resolver.TagResolver;
import org.spongepowered.configurate.ConfigurationNode;

import java.nio.file.Path;
import java.util.List;

public interface MessagesAPI {
    void create(Class<?> clazz, Path directory);
    void reload();

    ConfigurationNode getNode();
    String getString(String path);
    List<String> getStringList(String path);
    String getStringLore(String path);

    String getMessage(String path, TagResolver... resolvers);
    List<String> getMessageList(String path, TagResolver... resolvers);
    String getMessageLore(String path, TagResolver... resolvers);
}

File: src/main/kotlin/dev/tuhkanens/comfortlib/api/MessagesAPI.kt
package dev.tuhkanens.comfortlib.api

import net.kyori.adventure.text.minimessage.tag.resolver.TagResolver
import org.spongepowered.configurate.ConfigurationNode
import java.nio.file.Path

interface MessagesAPI {
    fun create(clazz: Class<*>, directory: Path)
    fun reload()
    fun getNode(): ConfigurationNode
    fun getString(path: String): String
    fun getStringList(path: String): List<String>
    fun getStringLore(path: String): String
    fun getMessage(path: String, vararg resolvers: TagResolver): String
    fun getMessageList(path: String, vararg resolvers: TagResolver): List<String>
    fun getMessageLore(path: String, vararg resolvers: TagResolver): String
}

inline fun <reified T : Any> MessagesAPI.create(directory: Path) =
    create(T::class.java, directory)